samples/subsys/usb/webusb/src/main.c

```diff
@@ -52,6 +55,18 @@ static struct msosv2_descriptor_t {
 		.dwWindowsVersion = 0x06030000,
 		.wTotalLength = sizeof(struct msosv2_descriptor_t),
 	},
+#if defined(CONFIG_USB_CDC_ACM)
+	/* If CONFIG_USB_CDC_ACM is selected, extra interfaces will be added on build time,
+	 * making the target a composite device, which requires an extra Function
+	 * Subset Header.
+	 */
+	.subset_header = {
+		.wLength = sizeof(struct msosv2_function_subset_header),
+		.wDescriptorType = MS_OS_20_SUBSET_HEADER_FUNCTION,
+		.bFirstInterface = 0, /* The WebUSB interface number */
```

Comment:
With this approach you actually need to update bFirstInterface to the actual value, like in CDC ACM implementation

https://github.com/zephyrproject-rtos/zephyr/blob/b83b9bede3bdc787615073a99463e6a00932318d/subsys/usb/device/class/cdc_acm.c#L451-L458

https://github.com/zephyrproject-rtos/zephyr/blob/b83b9bede3bdc787615073a99463e6a00932318d/subsys/usb/device/class/cdc_acm.c#L1186-L1188

but it would also require more rework, e.g. to move webusb.c code to main.c.
I suggest adding to the comment above that it only covers the CDC ACM use case, if other classes are enabled this sample may not work properly on Windows OS.

Reply:
I made more elaborate comments.  Ideally, we should probably update bFirstInterface as you mention, but I think it might be so application specific that the mechanism might need adjustments depending on the mix.

Reply:
I think updating the doc makes sense, but it should probably be a section on it's own related to the USB subsys on the MS specific headers and when to add what.